src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -61,15 +89,37 @@ public ResultTextMarker(int runIndex, Region region, string fullFilePath)
         /// <returns>Returns true if a VS editor was opened.</returns>
         public bool TryNavigateTo(bool usePreviewPane)
         {
-#pragma warning disable VSTHRD010 // This is an analyzer false positive. 
-            return this.TryNavigateTo(usePreviewPane, retryNaviation: true);
-#pragma warning restore VSTHRD010
-        }
-
-        private bool TryNavigateTo(bool usePreviewPane, bool retryNaviation)
-        {
             ThreadHelper.ThrowIfNotOnUIThread();
 
+            // If the tag doesn't have a persistent span, or it isn't open
+            // then this indicates that we need to attempt to open the document.
+            if (this.tag?.DocumentPersistentSpan.Span == null ||
+                !this.tag.DocumentPersistentSpan.IsDocumentOpen ||
+                this.tag.DocumentPersistentSpan.Span.TextBuffer == null)
+            {
+                if (!File.Exists(this.FullFilePath))
+                {
+                    if (!CodeAnalysisResultManager.Instance.TryRebaselineAllSarifErrors(runIndex, this.UriBaseId, this.FullFilePath))
+                    {
+                        return false;
+                    }
+                }
+                else if (!this.TryMapRegion())
+                {
+                    return false;
+                }
+
+                IVsWindowFrame vsWindowFrame = SdkUIUtilities.OpenDocument(ServiceProvider.GlobalProvider, this.FullFilePath, usePreviewPane);
+                if (vsWindowFrame == null)
+                {
+                    return false;
+                }
+
+                // This step is super important is it will cause the document to get "tagged" before the next check
```

Comment:
>step [](start = 24, length = 4)

window frame must be shown now (at this point) because we need tagging to occur before the rest of this method executes.